src/Microsoft.DotNet.Interactive.HttpRequestParser/Parser/HttpRequestParser.cs

```diff
@@ -336,18 +339,27 @@ private void ConsumeCurrentTokenInto(HttpSyntaxNode node)
                 ParseTrailingWhitespace(requestNode, stopAfterNewLine: true);
             }
 
-            var commentsToPrepend = new List<HttpCommentNode>();
+            var comments = new List<HttpCommentNode>();
             if (ParseComment() is { } comment)
             {
-                commentsToPrepend.Add(comment);
+                comments.Add(comment);
             }
 
             var headersNode = ParseHeaders();
             if (headersNode is not null)
             {
-                PrependCommentsIfAny(commentsToPrepend, headersNode);
+                AddCommentsIfAny(comments, headersNode);
                 requestNode.Add(headersNode);
             }
+            else
+            {
+                AddCommentsIfAny(comments, requestNode, addBefore: false);
+            }
+
+            if (ParseComment() is { } commentAfterHeaders)
```

Comment:
Does this work if there are multiple comments OR multiple comments separated by new lines?

Also looks like this won't handle any leading whitespace before `#` or `//`.

I still feel this is better handled via `ParseLeadingCommentsAndWhitespace` in the `Parse` method for whatever construct follows. However, I am yet to review your responses in the earlier PR - so please ignore if you already addressed this there :)

Reply:
One other thought is that perhaps `ParseTrailingWhitespace` can be updated to also parse trailing comments (and perhaps that will handle the case where `ParseLeadingCommentsAndWhitespace` can't handle the above comment (i.e., in the case where there is no construct that follows after the header section).

(Also wondering how we currently parse a document that only contains comments but no requests i.e., under what construct do such comments end up being added today?

It may be a good idea to add (skipped) tests for the above cases assuming we don't have them already.)

Reply:
We don't have a consistent pattern here but the same problems applies as we encountered with comments generally. Parsing leading whitespace in that way e.g. within `ParseComment` requires gathering it up and only adding it to the tree if we actually encounter a comment node. Otherwise, it needs to be stored and applied to the next node type that we do encounter.

Leading whitespace before comments is not supported in the old parser for either kind of comment prefix (`#` or `//`).

We don't support `//`-prefixed comments in the new parser at all.

Reply:
> We don't support //-prefixed comments in the new parser at all.

Based on the following code in the parser, `//` should be supported.
![image](https://github.com/dotnet/interactive/assets/10579684/1faa7fdc-eb47-4589-bbfd-e5eafc057150)

Reply:
The current design where we can only parse a single comment in these spots seems problematic. It is also odd that we don't support leading whitespace before comments just at these spots (we do support leading whitespace for any comments parsed via `ParseLeadingCommentsAndWhitespace` in other spots).

I can take a closer look and take a stab at improving this later. Will sync up offline once I have a PR / proposal.

Reply:
> Based on the following code in the parser, // should be supported.

This might work correctly now. I believe there were lookaheads in places that didn't take `//` into account.

Reply:
> The current design where we can only parse a single comment in these spots seems problematic.

This is just a bug. I haven't added combinatorial tests for all known syntax combinations in order to avoid trying to fix too many issues in a single massive PR. That and `//`-prefixed comments are up next.